src/Server/LspRequestRouter.cs

```diff
@@ -104,7 +104,7 @@ public async Task RouteNotification(IHandlerDescriptor descriptor, Notification
                         else
                         {
                             _logger.LogDebug("Converting params for Notification {Method} to {Type}", notification.Method, descriptor.Params.FullName);
-                            var @params = notification.Params.ToObject(descriptor.Params, _serializer.JsonSerializer);
+                            var @params = (notification.Params ?? new JObject()).ToObject(descriptor.Params, _serializer.JsonSerializer);
```

Comment:
@mholo65 this will probably fix the initialization issue, as it allows a null params though now that the IInitializedHandler has a type (it didn't used to).

Reply:
@david-driscoll, I’m afraid this isn’t related to the issue I’m seeing. The issue I’m having is a chicken-egg situation. I can’t create the handler until during initialize, because O# needs to know the target directory in order to createt the OmnisharpWorkspace. But in initialize, there’s no possibility to register the handlers because a) the factory method haven’t yet returned the LanguageServer instance and b) the IServiceCollection in Options used to instantiate the language server have already been built. In previous versions of this lib it was easier since we had access to the instance of the server inside our init delegate.